xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/test/java/org/xwiki/refactoring/script/RefactoringScriptServiceTest.java

```diff
@@ -363,4 +367,30 @@ public void changeDocumentAuthor() throws Exception
         verify(request).setReplaceDocumentAuthor(true);
         verify(request).setReplaceDocumentContentAuthor(true);
     }
+
+    @Test
+    void canSkipRecycleBinAllTrue()
+    {
+        when(this.documentAccessBridge.isAdvancedUser()).thenReturn(true);
+        when(this.configuration.canSkipRecyclebin()).thenReturn(true);
+        final boolean actual = this.refactoringScriptService.canSkipRecycleBin();
+        assertTrue(actual);
+    }
+
+    @Test
+    void canSkipRecycleBinConfigurationIsFalse()
+    {
+        when(this.configuration.canSkipRecyclebin()).thenReturn(false);
+        final boolean actual = this.refactoringScriptService.canSkipRecycleBin();
+        assertFalse(actual);
+    }
+
+    @Test
+    void canSkipRecycleBinAdvancedIsFalse()
+    {
+        when(this.documentAccessBridge.isAdvancedUser()).thenReturn(false);
+        when(this.configuration.canSkipRecyclebin()).thenReturn(true);
+        final boolean actual = this.refactoringScriptService.canSkipRecycleBin();
```

Comment:
You use a lot of final, we don't have this best practice. We need to decide if we want it or not. I feel it's not needed and it's making it harder to read as local variables.

Reply:
My bad, old habits die hard. Sorry. I'll remove them.

Reply:
Done.